spec/util/miq-xml_spec.rb

```diff
@@ -6,15 +6,34 @@
     expect(xml.root.elements[1].attributes['attr1']).to eq(attr_string)
   end
 
+  it "handles loaded document with top-level text nodes" do
+    attr_string = "test string"
+    doc_text = "XXX<test><element_1 attr1='#{attr_string}'/></test>"
+
+    xml = MiqXml.load(doc_text)
+    expect(xml.root.elements[1].attributes['attr1']).to eq(attr_string)
+
+    expect(xml.to_s).to start_with("XXX<test>")
+
+    xml.write(xml_str = '', 1)
+    expect(xml_str).to start_with("\n<test>")
```

Comment:
This is what I mentioned in the OP about `.write` no longer printing extraneous top-level text nodes.  I think this is because of https://github.com/ruby/rexml/pull/164 (although there are a number of parsing changes [between 3.3.1 and 3.3.2](https://github.com/ruby/rexml/compare/v3.3.1...v3.3.2))

Reply:
I also commented here about the weird leading blank link - That's probably a bug https://github.com/ruby/rexml/pull/164#discussion_r1683552024